### ISSUE TITLE
subprocess PIPE objects have finite buffer size

### ISSUE THREAD

**Issue by davidcox** [kanryu/pipeffmpeg#2]
Hello,

Me again.  I've been fiddling with pipeffmpeg to get it working, and I've run across a reasonably large problem.  subprocess PIPE objects have a limited buffer size, and once this buffer is overrun, the process will simply hang (without error).  Basically, this limits the allowable size of videos that can be accessed via this strategy.  Details on the larger issue can be found [here](http://www.popekim.com/2008/12/never-use-pipe-with-python-popen.html).

A different strategy is to read the frames one at a time, or in cached batches.  I've taken this strategy in https://github.com/davidcox/pipeffmpeg/tree/indexable_not_iterable

Basically, it computes time offsets from the video meta data and uses ffmpeg's `-ss` and `-vframes` options to pull out small batches of frames.

Thanks,
Dave
